### ISSUE TITLE
`nearby_ids` allocation

### ISSUE THREAD

**Issue by AayushSabharwal** [JuliaDynamics/Agents.jl#617]
`nearby_ids` (and functions that rely on it) are highly allocating and possibly a major bottleneck for performance. Consider this flocking benchmark, which uses a minimally-allocating scheduler from #613:
```julia
julia> @benchmark step!($(now), as, 1000)
BenchmarkTools.Trial: 6 samples with 1 evaluation.
 Range (min … max):  829.740 ms …    1.542 s  ┊ GC (min … max): 14.54% … 23.66%
 Time  (median):     882.250 ms               ┊ GC (median):    17.00%
 Time  (mean ± σ):   993.958 ms ± 274.066 ms  ┊ GC (mean ± σ):  18.01% ±  5.61%

  ▁█     ▁    ▁                                               ▁
  ██▁▁▁▁▁█▁▁▁▁█▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁█ ▁
  830 ms           Histogram: frequency by time          1.54 s <

 Memory estimate: 970.84 MiB, allocs estimate: 11951443.
```

Nearly 1GB allocated, basically just from `nearby_ids` in the step function.

**Comment by AayushSabharwal** [comment]
> What is `now`?

> What is `as`?

Ah, sorry, I copied the benchmark over from #613. `now` is simply an instance of the flocking model, and `as` the step function.

> nearby_ids is totally different depending on the space used.

I forgot about that, oops. Flocking uses `ContinuousSpace`, so that one.

**Comment by Datseris** [comment]
We do calculate `nearby_ids` in our benchmark suite: https://github.com/JuliaDynamics/Agents.jl/blob/master/benchmark/benchmarks.jl#L17 

So you can just run these. We should also start actually storing these benchmarks somewhere as json or csv, probably in the ABM_FRAMEWORK_COMPARISON repo. The benchmarks use PkgBenchmark, it is really straightforward to use, I'm sure you'll learn it in 5 minutes or so.  https://github.com/JuliaCI/PkgBenchmark.jl

**Comment by AayushSabharwal** [comment]
For some reason, the benchmark is stuck on `SUITE["graph"]["add_union"]["agent_single"]`

**Comment by Datseris** [comment]
Then that's probably a bug that needs to be fixed. I forgot to say, these benchmarks _have not_ been uypdated to version 5 of agents.

**Comment by Datseris** [comment]
Closed in #636
